### PR TITLE
test: Add E2E tests to validate pod grouping and waiting fix

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

**How was this change tested?**

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: test/suites/termination/termination_test.go [hunk 1]
```diff
@@ -47,4 +51,112 @@ var _ = Describe("Termination", func() {
 			g.Expect(lo.FromPtr(env.GetInstanceByID(instanceID).State.Name)).To(BeElementOf("terminated", "shutting-down"))
 		}, time.Second*10).Should(Succeed())
 	})
+	// Pods from Karpenter nodes are expected to drain in the following order:
+	//   1. Non-Critical Non-Daemonset pods
+	//   2. Non-Critical Daemonset pods
+	//   3. Critical Non-Daemonset pods
+	//   4. Critical Daemonset pods
+	// Pods in one group are expected to be fully removed before the next group is executed
+	It("should drain pods on a node in order", func() {
```
Comment (jmdeal) [inline test/suites/termination/termination_test.go:60]: Should we also include `system-cluster-critical` pods and daemonsets like in the functional tests upstream?

File: test/pkg/environment/common/expectations.go [hunk 2]
```diff
@@ -329,6 +329,17 @@ func (env *Environment) ConsistentlyExpectTerminatingPods(duration time.Duration
 	}, duration.String()).Should(Succeed())
 }
 
+func (env *Environment) ConsistentlyExpectLivePods(duration time.Duration, pods ...*corev1.Pod) {
```
Comment (jmdeal) [inline test/pkg/environment/common/expectations.go:332]: nit: do we want to rename this to `ConsistentlyExpectActivePods` to match the upstream nomenclature?

**Review by jmdeal — COMMENTED** [review]
A couple of small comments, looks good overall

File: test/suites/termination/termination_test.go (hunk 1 above)
Comment (jonathan-innis) [reply test/suites/termination/termination_test.go:60]: Good callout -- will add

File: test/pkg/environment/common/expectations.go (hunk 2 above)
Comment (jonathan-innis) [reply test/pkg/environment/common/expectations.go:332]: Sure -- sounds fine to me.

**Review by jmdeal — APPROVED** [review]
LGTM 🚀

**Pull request merged**